### PR TITLE
Implemented Terrain Layer Spawner and Height/Surface Gradient List wrapped nodes

### PR DESCRIPTION
**Pull request by cgalvan**
## What does this PR do?

Added support for the Terrain Layer Spawner node, as well as Height and Surface Gradient List wrapped nodes that extend the spawner. As part of this PR had to make some modifications to Landscape Canvas that assumed there would only be Vegetation wrapped nodes.

TerrainLayerSpawnerAndHeightGradientList

## How was this PR tested?

Ran Landscape Canvas automated tests. Also tested the vegetation wrapper nodes to make sure they still function as expected.

### CODE REVIEW

File: Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainLayerSpawnerNode.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+// Graph Model
+#include <GraphModel/Integration/Helpers.h>
+#include <GraphModel/Model/Slot.h>
+
+// Landscape Canvas
+#include "TerrainLayerSpawnerNode.h"
```
Comment (mbalfour-amzn) [inline Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainLayerSpawnerNode.cpp:23]: Nit: Ought to be `#include <Editor/Nodes/Terrain/TerrainLayerSpawnerNode.h>`

File: Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainSurfaceGradientListNode.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// AZ
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+// Graph Model
+#include <GraphModel/Integration/Helpers.h>
+#include <GraphModel/Model/Slot.h>
+
+// Landscape Canvas
+#include "TerrainSurfaceGradientListNode.h"
```
Comment (mbalfour-amzn) [inline Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainSurfaceGradientListNode.cpp:21]: Nit: Should be `#include <Editor/Nodes/Terrain/TerrainSurfaceGradientListNode.h>`

**Review by mbalfour-amzn — APPROVED** [review]

File: Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainLayerSpawnerNode.cpp (hunk 1 above)
Comment (cgalvan) [reply Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainLayerSpawnerNode.cpp:23]: Thanks will fix!

File: Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainSurfaceGradientListNode.cpp (hunk 2 above)
Comment (cgalvan) [reply Gems/LandscapeCanvas/Code/Source/Editor/Nodes/Terrain/TerrainSurfaceGradientListNode.cpp:21]: Thanks will fix!

File: Gems/LandscapeCanvas/Code/Source/Editor/Core/Core.h [hunk 3]
```diff
@@ -44,6 +44,7 @@ namespace LandscapeCanvas
     static const QString GRADIENT_GENERATOR_TITLE = QObject::tr("Gradient");
     static const QString GRADIENT_MODIFIER_TITLE = QObject::tr("Gradient Modifier");
     static const QString VEGETATION_AREA_TITLE = QObject::tr("Vegetation Area");
+    static const QString TERRAIN_TITLE = QObject::tr("Terrain");
```
Comment (pollend) [inline Gems/LandscapeCanvas/Code/Source/Editor/Core/Core.h:47]: these probably won't translate correctly if the language is changed at any point.

you probably want to do something like this: https://doc.qt.io/qt-6/i18n-source-translation.html#using-qt-tr-noop-and-qt-translate-noop-in-c

**Review by gadams3 — APPROVED** [review]

**Pull request merged**